### PR TITLE
Part1

### PR DESCRIPTION
**Pull request by DvDream**
Here's my pull request for Part 1a-1d assignments

### CODE REVIEW

**Review by matkatmusic — CHANGES_REQUESTED** [review]
Cool.  Move on to the next set of instructions in the project!

File: main.cpp [hunk 1]
```diff
@@ -1010,7 +901,375 @@ Part 1e - Step 19: Request a review
...
+    struct Train
+    {
+        int numberOfWagons = 12;
+        int numberOfSeats = 120;
+        int trainId = 123;
+        bool hasArrived = false;
+        float fuelPercentage = 100.0f;
+
+        void leaveStation(float fuelPercentage, bool setAutoPilot = false); // Function to indicate that the train has to leave the station. It accepts fuelPercetnage parameter to check that it can actually leave and if it need to turn the Autopilot on. 
+        double DayMilesTraveled(std::string day); // return the mailes traveled in a given day
```
Comment (matkatmusic) [inline main.cpp]: This is a bad function name.  The function name does not match the description.
`computeMilesTraveledPerDay` is a much more accurate function name.
Also, the coding style does not conform to the course coding standard, which is detailed in the Readme.MD. 
- [x] revise
- [x] review the course coding standard from the Readme.MD

File: main.cpp [hunk 2]
```diff
@@ -1010,7 +901,375 @@ Part 1e - Step 19: Request a review
...
+    int speedIndication = 0;
+    //    4) fuel indicator in percentage (int)
+    int fuelPercentage = 50;
+    //    5) hour indicator (std::string)
+    std::string hourIndication = "00:00:00";
+    //3 things it can do:
+    //    1) accelerate
+    int accelerate(int accelerationAmount); // return an updated speedIndication value given an acceleration amount as parameter
+    //    2) brake
+    int brakeOfAGivenAmount(int brakingAmount); // return an updated speedIndication value given braking amount as parameter
```
Comment (matkatmusic) [inline main.cpp]: This function name makes no sense
- [x] revise

File: main.cpp [hunk 3]
```diff
@@ -1010,7 +901,375 @@ Part 1e - Step 19: Request a review
...
+    int numberOfSeats = 1;
+    //    4) colour (std::string)
+    std::string colour = "black";
+    //    5) percentage of cushioning (float)
+    float cushioningPercentage = 90.0f;
+    //3 things it can do:
+    //    1) absorb vibrations
+    void absorbVibrations();
+    //    2) cover objects holder
+    void coverObjectsHolder();
```
Comment (matkatmusic) [inline main.cpp]: How does a seat cover an object holder?
What is an object holder?
Pick better actions for a seat that make sense.  
- [x] revise

File: main.cpp [hunk 4]
```diff
@@ -1010,7 +901,375 @@ Part 1e - Step 19: Request a review
...
+    int degreesAngulation = 90;
+    //    5) type of light (std::string)
+    std::string type = "halogen";
+    //3 things it can do:
+    //    1) illuminate street
+    double illuminateStreet(double brightnessAmount); // Based upon the brightness amount, it returns the updated brightness percentage
+    //    2) alert people
+    void alertPeople();
+    //    3) indicate change of direction
+    std::string changeDirection(int direction); // direction could be 1 or 2 to indicate left or right. It returns the direction as a string.
```
Comment (matkatmusic) [inline main.cpp]: the action is 'Indicate", not "Change"
the current function name is therefore incorrect.
- [x] revise

**Review by matkatmusic — CHANGES_REQUESTED** [review]

File: main.cpp (hunk 2 above)
Comment (DvDream) [reply main.cpp]: changed function name and description

File: main.cpp (hunk 3 above)
Comment (DvDream) [reply main.cpp]: changed action, hoping the new one makes more sense than the last one.

**Review by matkatmusic — APPROVED** [review]

**Pull request merged**